libs/data-mapper/src/lib/core/services/appInsights/AppInsights.ts

```diff
@@ -0,0 +1,17 @@
+import { ReactPlugin } from '@microsoft/applicationinsights-react-js';
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+
+export const reactPlugin = new ReactPlugin();
+
+const appInsights = new ApplicationInsights({
+  config: {
+    connectionString:
```

Comment:
Can we insert this during build time and just disable tracking if it's missing. Generally open source projects built from source should not track by default.

Reply:
Lets also make sure there is a vscode setting or something that can turn it off.

Reply:
There is this [disabling option](https://code.visualstudio.com/docs/getstarted/telemetry#_disable-telemetry-reporting) in the vscode configuration. Deep down these libraries take a look at it and decide whether log it or not. @NicL9923 just to double check try it out.

Reply:
@ccastrotrejo So then that setting should be enough huh? If the package we use for that in the VSIX automatically utilizes it under the hood?

Reply:
Yep, that should be fine. Thats the way turn it on/off.

Reply:
@hartra344 @ccastrotrejo It looks like the React AI package plays nice (no thrown exceptions or console errors) if you just don't make the loadAppInsights call but still make the trackX calls as normal - so that should be good - gonna follow up about plugging in the keys in Teams